models/collectionPoint.js

```diff
@@ -1,29 +1,50 @@
-'use strict';
+"use strict";
 
 const Event = require("./event");
-const User = require("./user")
+const User = require("./user");
 
 module.exports = (sequelize, DataTypes) => {
-  const collectionPoint = sequelize.define('collectionPoint', {
-    name: DataTypes.STRING,
-    eventId: DataTypes.INTEGER
-    //TODO: Add Location (coordinate)
-  }, {});
-  collectionPoint.associate = function(models) {
+  const collectionPoint = sequelize.define(
+    "collectionPoint",
+    {
+      name: DataTypes.STRING,
+      eventId: DataTypes.INTEGER,
+      //TODO: Add Location (coordinate)
+    },
+    {
+      paranoid: true,
+      hooks: {
+        afterDestroy: function (instance, options) {
+          instance.getPatients().then((patients) =>
+            patients.map((patient) => {
+              patient.destroy();
+            })
+          );
+        },
+        afterRestore: function (instance, options) {
+          instance
+            .getPatients({ paranoid: false })
+            .then((patients) => patients.map((patient) => patient.restore()));
```

Comment:
Good question. Correct me if I am wrong @petorsong but we had discussed this exact case. On the front-end, there is no functionality to system-delete a patient. Instead, the user will have the option to change the status of a patient to `DELETED`. This means that the only way to system-delete a patient is if we delete a collection point. If we didn't do this, then you would be right: we would have no past knowledge whether we deleted a patient through a cascade or manually deleted a single patient.

Reply:
yep the reason we don't want to use the actual soft delete for patients is because we wanted to avoid this ambiguity between deleted patients and patients with the `DELETED` status - we will only be using the latter for now, but also have the former available if we do want to use it later on

Reply:
Ah okay, thanks for the clarification.